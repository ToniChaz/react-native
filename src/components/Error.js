/**
 * Weather App
 * Build with react-native
 * @error-weather
 */

import React, { Component } from 'react'
import { StyleSheet, Text, View } from 'react-native'

export default class Error extends Component {

	render() {
		const {error} = this.props

		return (
			<View style={styles.errorTextView}>
				<Text style={styles.errorText}>{error.message}</Text>
			</View>
		)
	}
}

const styles = StyleSheet.create({
	errorText: {
		color: '#FF0000'
	},
	errorTextView: {
		marginTop: 20,
		marginBottom: 10,
		justifyContent: 'center',
		alignItems: 'center'
	}
})