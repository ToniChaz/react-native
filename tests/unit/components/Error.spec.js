// Error.spec.js

import 'react-native'
import React from 'react'
import Error from '../../../src/components/Error.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

describe('Error component', () => {
	test('renders correctly', () => {
		const error = {
			message: 'The custom error message'
		}

		const tree = renderer.create(
			<Error error={error} />
		)

		expect(tree.toJSON()).toMatchSnapshot()
	})
})