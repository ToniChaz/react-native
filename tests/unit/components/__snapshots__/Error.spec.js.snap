// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 10,
      "marginTop": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#FF0000",
      }
    }
  >
    The custom error message
  </Text>
</View>
`;
